### ISSUE TITLE
How to make it connect?

### ISSUE THREAD

**Issue by mikekenyon99** [SchoofsKelvin/vscode-sshfs#124]
Hi,

Using Visual Code 1.32.3 on macOS and installed this extension.  I can connect to my SSH through Terminal (ssh pi@MikePiZero_2) but using extension I can't get it to connect.  I keep getting a:

![Screenshot 2019-03-31 at 18 16 10](https://user-images.githubusercontent.com/42475982/55292314-106f7e00-53e1-11e9-98cd-e41ea7f4acec.png)

Where am I going wrong?

settings.json as below:

![Screenshot 2019-03-31 at 18 17 41](https://user-images.githubusercontent.com/42475982/55292331-47459400-53e1-11e9-921f-e3ce6b9f3130.png)

Thanks

Mike

**Comment by SchoofsKelvin** [comment]
The `host` field should either be an IP address, or a domain the extension will do a dns lookup on. I assume you defined `MikePiZero_2` in `.ssh/config` or something similar? Currently, those files are ignored.

**Comment by mikekenyon99** [comment]
Thanks it worked when I used the IP.  I just assumed by "hostname" I could use the dns name. 


![Screenshot 2019-04-01 at 21 16 44](https://user-images.githubusercontent.com/42475982/55356795-76cccd00-54c3-11e9-83bb-354686b246ab.png)

**Comment by WSLUser** [comment]
Have you tried the FQDN? That's what I use for my server and it connects successfully without a .ssh/config (though it would be nice to use it).

**Comment by mikekenyon99** [comment]
Sorry, Im not a network expert, I don't know how to turn MikePiZero_2 into a FQDN. I just normally use ssh pi@MikePiZero_2 at my terminal and all works.  Sorry if Im being stupid.

**Comment by WSLUser** [comment]
If it's on a domain, it should have a @sub-domain.domain like github.com. If it's local then you need to go to system name and get the full name from there. On Mac, it should have .local (verify full name by going to System Preferences/Sharing). Windows is a bit different but can be easily found by right clicking on Computer from Start Screen and choosing Properties. Hope this helps.

**Comment by mikekenyon99** [comment]
Thank you for your help, but now I'm totally confused.  Went to System Preferences/Sharing and its all about sharing stuff on my Mac.  Whats there relevance?  Are you saying I have to open shared folders on my Mac to SSH fro Visual Studio?

**Comment by WSLUser** [comment]
At the top, your computer name is listed. You'll see at the top: 

> Computer Name: .....
Computers on your local network can access your computer at:
Full name including .local (or whatever)

Use the full name with .local

**Comment by mikekenyon99** [comment]
Ah, I'm with you now.  However that doesn't fix the issue with this plugin.  I run Visual Studio on the Mac and I want to access the SSH session on the Pi.  Not the SSH on the Mac.  However, having read up on host files now I would need to enter that into the ~/ssh/config and anyhow this extension doesn't read it so its IP addresses all the way.

**Comment by WSLUser** [comment]
Should be the same process of getting the full system name from the Pi. If that doesn't work, then I'd say there's an edge case with DNS not working with Raspberry Pis.

**Comment by mikekenyon99** [comment]
Think it's an edge case, would need an entry in the config file, below is what is reported and I have tried that. 
![Screenshot 2019-04-02 at 19 43 09](https://user-images.githubusercontent.com/42475982/55427753-9aede400-557f-11e9-8907-b134ef6792fb.png)

**Comment by WSLUser** [comment]
Ok last shot after a quick google since uname usually enough for machine name. If you put in 

> hostname

That should get you the FQDN.

**Comment by mikekenyon99** [comment]
Yeah that's the output.  Never mind I can use the IP address. :-) 

![Screenshot 2019-04-02 at 20 32 08](https://user-images.githubusercontent.com/42475982/55430618-7812fe00-5586-11e9-8c30-7e67ce752780.png)

**Comment by SchoofsKelvin** [comment]
I'm just using node's [dns.resolve](https://nodejs.org/docs/latest-v10.x/api/dns.html#dns_dns_resolve_hostname_rrtype_callback), which, reading the [implementation considerations](https://nodejs.org/docs/latest-v10.x/api/dns.html#dns_dns_resolve_dns_resolve_and_dns_reverse), always does an actual DNS lookup through servers, ignoring things like `/etc/hosts`.

It seems like I should use `dns.lookup` instead, to support local hostnames.
A quick test on my PC (Windows 10 with hostname `WWW`) provide these results:
```js
// Those three give the same results as `nslookup`
// (`nslookup` reports these are resolved by my router at 192.168.1.1)
// 192.168.1.11 is indeed my PC's IP on my local LAN (with the router as gateway)
dns.resolve('WWW') // 192.168.1.11
dns.resolve('WWW.home') // 192.168.1.1
dns.resolve('WWW.local') // ENOTFOUND

// 172.29.25.145 is my IP on some Hyper-V Virtual Adapter, so technically a correct lookup
dns.lookup('WWW') // 172.29.25.145
dns.lookup('WWW.home') // 192.168.1.11
    // ^ most likely resulted in a real DNS request to my router
dns.lookup('WWW.local') // 172.29.25.145

// The above repeated, but with the virtual adapter disabled first
dns.lookup('WWW') // 192.168.1.11
dns.lookup('WWW.home') // 192.168.1.11
dns.lookup('WWW.local') // 192.168.1.11
```

I'll still wait a few days for feedback here, but it seems like I should switch from `dns.resolve` to `dns.lookup`. This would support local hostnames. *During my testing `dns.resolve` kind of did support local hostnames, but this is most likely because of my DNS server (my router) resolving them.*

**Comment by SchoofsKelvin** [comment]
Switched from `dns.resolve` to `dns.lookup` (4937028), now live in version 1.15.2 of the extension.
